src/request-extractors/hapi.ts

```diff
@@ -77,22 +77,22 @@ function extractRemoteAddressFromRequest(req: hapi.Request) {
  * @returns {RequestInformationContainer} - an object containing the request
  *  information in a standardized format
  */
-export function hapiRequestInformationExtractor(req: hapi.Request) {
+export function hapiRequestInformationExtractor(req?: hapi.Request) {
   const returnObject = new RequestInformationContainer();
 
-  if (!isObject(req) || !isObject(req.headers) || isFunction(req) ||
+  if (!isObject(req) || !isObject(req!.headers) || isFunction(req) ||
```

Comment:
Same question.

Reply:
I don't want to change the use of `isObject` for the reasons stated in previous comments.  However, `isObject(undefined) = false` and so it is safe to use the `!` annotation in `!isObject(req!.headers)`.

Reply:
But isn't `isObject(null) === true`? I usually try to avoid `!` as much as possible because it can cause programming errors when changing existing code without thinking very closely about nullability.

If you're planning to create a separate PR for removing `is`, I think this is fine for now.

Reply:
Can you prepend `!req || ...` to this line to ensure that `req` is henceforth non-undefined?

Reply:
I think it would be best to make these changes in a separate PR that removes the use of `is*` functions.